federation/pkg/kubefed/init/init_test.go

```diff
@@ -123,13 +139,22 @@ func TestInitFederation(t *testing.T) {
 		if "" != tc.etcdPVCapacity {
 			cmd.Flags().Set("etcd-pv-capacity", tc.etcdPVCapacity)
 		}
+		if "valid-run" == tc.dryRun {
+			cmd.Flags().Set("dry-run", "true")
+		}
+
 		cmd.Run(cmd, []string{tc.federation})
 
 		if tc.expectedErr == "" {
 			// uses the name from the federation, not the response
 			// Actual data passed are tested in the fake secret and cluster
 			// REST clients.
-			want := fmt.Sprintf("Federation API server is running at: %s\n", tc.lbIP)
+			want := ""
+			if "" != tc.dryRun {
```

Comment:
You are not dropping this because we decided to keep the outputs different, right?

Also, can you write the comparison in the reverse order. I mean, as `tc.dryRun != ""` Also, drop the unnecessary `else`. You can write this whole thing as:

```go
want := fmt.Sprintf("Federation API server is running at: %s\n", tc.lbIP)
if tc.dryRun != "" {
    want = fmt.Sprintf("Federation control plane runs (dry run)\n")
}
```